blog(embertimes165): rfc deprecate tryinvoke (#786)
ember-learn/ember-blog: source/2021-10-09-the-ember-times-issue-165.md

---

## [RFC: Deprecate tryInvoke 📜](https://github.com/emberjs/rfcs/pull/673)

> [Bach Vo (@bachvo)](https://github.com/bachvo) created a new RFC to [Deprecate tryInvoke](https://github.com/emberjs/rfcs/pull/673) building on top of the previous [RFC to deprecate getWithDefault](https://emberjs.github.io/rfcs/0554-deprecate-getwithdefault.html).

Reviewer: ```suggestion
[Bach Vo (@bachvo)](https://github.com/bachvo) created a new RFC to [deprecate tryInvoke](https://github.com/emberjs/rfcs/pull/673). It builds on top of the previous [RFC to deprecate getWithDefault](https://emberjs.github.io/rfcs/0554-deprecate-getwithdefault.html).
```

> This is because in most cases, function arguments should not be optional, but in the rare occasion that it is optional, the Javascript language has [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) so we can deprecate the usage of `tryInvoke`.

Reviewer: Since it's an RFC, I thought it may be good to state that the following sentences are the author's opinion.

```suggestion
The author explains that, in most cases, function arguments should not be optional. In the rare occasion that it is optional, the Javascript language has [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) so we can deprecate the usage of `tryInvoke`.
```

Author: ya this is fair

> Before:
> ```js
> import { tryInvoke } from '@ember/utils';
>  
> foo() {
>  tryInvoke(this.args, 'bar', ['baz']);
> }
> ```
> 
> After:
> ```js
> foo() {
>  this.args.bar?.('baz');
> }
> ```

Reviewer: Linting required adding a line between code blocks.

``````suggestion
Before:

```js
import { tryInvoke } from '@ember/utils';
 
foo() {
 tryInvoke(this.args, 'bar', ['baz']);
}
```

After:

```js
foo() {
 this.args.bar?.('baz');
}
```
``````
